sdk/typescript/src/signers/txn-data-serializers/txn-data-serializer.ts

```diff
@@ -60,7 +60,30 @@ export interface MoveCallTransaction {
 }
 
 export interface PublishTransaction {
-  compiledModules: string[];
+  /**
+   * Transaction type used for publishing Move modules to the Sui.
+   * Should be already compiled using `sui-move`, example:
+   * ```
+   * $ sui move build
+   * $ cat build/project_name/bytecode_modules/module.mv
+   * ```
+   * In JS:
+   *
+   * ```
+   * // If you are using `RpcTxnDataSerializer`,
+   * let file = fs.readFileSync('./move/build/project_name/bytecode_modules/module.mv', 'base64');
+   * let compiledModules = [file.toString()]
+   *
+   * // If you are using `LocalTxnDataSerializer`,
+   * let file = fs.readFileSync('./move/build/project_name/bytecode_modules/module.mv');
+   * let modules = [ Array.from(file) ];
+   *
+   * // ... publish logic ...
+   * ```
+   *
+   * Each module should be represented as a sequence of bytes.
+   */
+  compiledModules: Iterable<string> | Iterable<Iterable<number>>;
```

Comment:
why this can be `Iterable<Iterable<number>>`?

Reply:
`Iterable<number>` represents an array of bytes